### PR TITLE
added possibility to change shortcut in shortcut details

### LINKED ISSUE #1044: add possibility to change shortcut from GUI
For example, `WINEDEBUG` is currently always set to `-all`. It should be possible to change this from the GUI. The same applies to the arguments and almost any other field in the `ShortcutDTO#script` (maybe except of `type`).

### PR DESCRIPTION
**Pull request by plata**
You can open the new panel to edit a shortcut by right-clicking an item in your library and then selecting "Edit" from the context menu.

fixes #1044 

change

### CODE REVIEW

**Comment by madoar** [comment]
I would prefer a save button instead of automatically saving changes.

**Comment by plata** [comment]
Maybe it would be better to have a separate panel to edit a shortcut then? This would allow us to edit also miniature etc.

**Comment by madoar** [comment]
Yes I think so too.

**Comment by madoar** [comment]
How do you enter the new shortcut edit panel?
Can you provide some screenshots how this looks like?
About the "Save" button, I think it should have a border, to mark it visibly as a button. When I see your screenshot I could think that it is a normal label.

**Comment by plata** [comment]
Like I wrote in the description. There's a new context menu.
context menu
It isn't transparent normally. That just happened when taking the screenshot.

The "Save" button looks like any other button in the Breeze Dark theme. We can think about changing that but it's clearly not belonging to this PR.

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.phoenicis.configuration.localisation.Localisation.tr;
+
+final class EditShortcutPanel extends DetailsView {
+    private final Logger LOGGER = LoggerFactory.getLogger(EditShortcutPanel.class);
+    private static final String CAPTION_TITLE_CSS_CLASS = "captionTitle";
+
+    ShortcutDTO editedShortcut;
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java]: private

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.phoenicis.configuration.localisation.Localisation.tr;
+
+final class EditShortcutPanel extends DetailsView {
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java]: JavaDoc

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        GridPane buttons = new GridPane();
+        buttons.setAlignment(Pos.CENTER);
+
+        ColumnConstraints runColumn = new ColumnConstraints();
+        runColumn.setPercentWidth(30);
+        ColumnConstraints stopColumn = new ColumnConstraints();
+        stopColumn.setPercentWidth(30);
+        ColumnConstraints uninstallColumn = new ColumnConstraints();
+        uninstallColumn.setPercentWidth(30);
+        buttons.getColumnConstraints().addAll(runColumn, stopColumn, uninstallColumn);
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java]: 30% * 3 = 90% what about the missing 10%?

File: phoenicis-javafx/src/main/resources/org/phoenicis/javafx/themes/breezeDark/main.less [hunk 4]
```diff
@@ -87,6 +87,19 @@
     -fx-background-color: @focus-color;
 }
 
+.context-menu {
+  -fx-background-color: @background-color;
+  -fx-text-fill: @text-color;
+}
+
+.menu-item:hover {
```
Comment (madoar) [inline phoenicis-javafx/src/main/resources/org/phoenicis/javafx/themes/breezeDark/main.less]: Aren't the menu-items contained inside the context-menu?

Comment (plata) [inline phoenicis-javafx/src/main/resources/org/phoenicis/javafx/themes/breezeDark/main.less]: Doesn't seem so.

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/ViewLibrary.java [hunk 5]
```diff
@@ -66,22 +70,34 @@
...
 
-            availableShortcuts.deselectAll();
-            availableShortcuts.select(selectedItem);
-            onShortcutSelected.accept(selectedItem);
-            showShortcutDetails(selectedItem);
+        availableShortcuts = new CombinedListWidget<>(ListWidgetEntry::create, (selectedItem, event) -> {
 
-            if (event.getClickCount() == 2) {
-                onShortcutDoubleClicked.accept(selectedItem);
+            if (event.getButton() == MouseButton.PRIMARY) {
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/ViewLibrary.java]: Can you add some short comments what's happening in these two if blocks?

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                final Label keyLabel = new Label(tr(unCamelize(entry.getKey())) + ":");
+                keyLabel.getStyleClass().add(CAPTION_TITLE_CSS_CLASS);
+                GridPane.setValignment(keyLabel, VPos.TOP);
+                this.gridPane.add(keyLabel, 0, i);
+
+                final TextArea valueLabel = new TextArea(entry.getValue().toString());
+                valueLabel.setWrapText(true);
+                valueLabel.setPrefRowCount(entry.getValue().toString().length() / 25);
+                valueLabel.focusedProperty().addListener((observable, oldValue, newValue) -> {
+                    // if TextArea looses focus
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java]: what then?

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/ViewLibrary.java [hunk 7]
```diff
@@ -171,6 +188,13 @@ private void showShortcutDetails(ShortcutDTO shortcutDTO) {
         this.showDetailsView(libraryPanel);
     }
 
+    private void showShortcutEdit(ShortcutDTO shortcutDTO) {
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/ViewLibrary.java]: JavaDoc

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        this.description = new Label();
+        this.description.setWrapText(true);
+
+        this.gridPane = new GridPane();
+        this.gridPane.getStyleClass().add("grid");
+        this.gridPane.getColumnConstraints().addAll(new ColumnConstraintsWithPercentage(30),
+                new ColumnConstraintsWithPercentage(70));
+
+        Region spacer = new Region();
+        spacer.setPrefHeight(40);
```
Comment (madoar) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/EditShortcutPanel.java]: Can we set the spacer in css?

**Pull request merged**